karapuzio/assignment3/WordSearch.java

```diff
@@ -0,0 +1,141 @@
+package edu.codeU.assignment3;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * The class to find all words in the grid.
+ */
+public class WordSearch {
+    /**
+     * The number of rows in grid.
+     */
+    private int rows;
+    /**
+     * The number of columns in grid;
+     */
+    private int columns;
+    /**
+     * The grid in which we look words.
+     */
+    private char[][] grid;
+    /**
+     * The List of Words in dictionary.
+     */
+    private List<String> dict;
+    /**
+     * Flag array that use in dfs to check existing of current cell of grid at current string.
+     */
+    private boolean flag[][];
+
+    public WordSearch(int rows, int columns, char[][] grid, List<String> dict) {
+        this.rows = rows;
+        this.columns = columns;
+        this.grid = grid;
+        this.dict = dict;
+        flag = new boolean[rows][columns];
+    }
+
+    /**
+     * We find x, y - coordinates in grid of all neighbors.
+     * @param x is the current row.
+     * @param y is the current column.
+     * @param neighborX the x - coordinates of all neighbors.
+     * @param neighborY the y - coordinates of all neighbors.
+     */
+    private void getAllNeighbors(int x, int y, List<Integer> neighborX, List<Integer> neighborY){
+        int SIZE_DELTAS = 3;
+        int[] deltaX = {-1, 0, 1};
+        int[] deltaY = {-1, 0, 1};
+        for (int i = 0; i < SIZE_DELTAS; i++){
+            for (int j = 0; j < SIZE_DELTAS; j++){
+                if ( (deltaX[i] != 0  || deltaY[j] != 0) && x+deltaX[i] >= 0 && x+deltaX[i] < rows && y+deltaY[j] >= 0 && y+deltaY[j] < columns){
+                    neighborX.add(x+deltaX[i]);
+                    neighborY.add(y+deltaY[j]);
+                }
+            }
+        }
+    }
+
+     /**
+     * The method to make traversal in the grid starting at position (x,y) and current word curWord.
+     * If we find a word we add it ti words set.
+     * We don't go far in graph if current string are not prefix of any word in dictionary.
+     * @param x is current rows.
+     * @paran y pd current column.
+     * @param curWord is the current word that was formed before in traversal.
+     * @param words is the set that contains all words that we found at the current moment.
+     */
+    private void dfs(int x, int y, String curWord, Set<String> words){
+        flag[x][y] = true;
+        char curChar = grid[x][y];
```

Comment:
You don't really need this variable, just add the value directly to the string in the line below